packages/@ember/-internals/glimmer/lib/components/link-to.ts

```diff
@@ -891,11 +878,8 @@ const LinkComponent = EmberComponent.extend({
     this.set('queryParams', queryParams);
 
     // 4. Any remaining indices (excepting `targetRouteName` at 0) are `models`.
-    if (params.length > 1) {
-      this.set('models', this._getModels(params));
-    } else {
-      this.set('models', []);
-    }
+    params.shift();
```

Comment:
We should not mutate the params passed in to us

Reply:
@rwjblue they are sliced at the very top of the function

Reply:
Doh! thank you @Turbo87